webhook_to_fedora_messaging/config/config.py

```diff
@@ -0,0 +1,20 @@
+# SPDX-FileCopyrightText: Contributors to the Fedora Project
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+import sys
+from os import environ
+
+import toml
+
+
+def return_flaskapp_config_data():
+    try:
+        path = environ["W2FM_APPCONFIG"]
+        data = toml.load(path)["flaskapp"]
+        return data
+    except FileNotFoundError:
+        sys.exit(1)
```

Comment:
Let's not use `sys.exit()` here. The only places where `sys.exit()` is appropriate are CLI scripts, because they know exactly the context in which they are called.
When writing code, always have in mind how you are going to test it. Calling `sys.exit()` will make it harder.

Reply:
Hein? 🤨 

https://github.com/gridhead/syncstar/blob/main/test/test_isos.py#L99-L101

Reply:
Yeah, of course there are ways to do it, you could also mock the `sys.exit` function in the tests. There are always ways to do things in Python ;-)
The first remark still stands. A regular function should not call `sys.exit()` unless it's a CLI. We should catch this exception (or a custom `ConfigError` exception) in the place where we know how to print the error message and handle the behavior (a CLI would exit with a return code, a webhook would raise a 500 error, etc)

Reply:
Just so I understand it correctly, do you want for me to raise a custom exception like `ConfigError` in the block https://github.com/fedora-infra/webhook-to-fedora-messaging/pull/19/files#diff-8bcb49b371beb99197d8db606d6491dcbdc1cdaa9d2e3c3a1772b0f743b68271R18-R20 and the exception function can include things like `sys.exit()` and `logging.error` or do you have something else in your mind?

Reply:
Yes, if you mean what I think you mean by "exception function" :-)

My thinking is that the `config.py` module does not know how to best handle a config error. Sometimes it's quitting, sometimes it's returning an error, sometimes it should be printed to stdout, sometimes not, etc. Raising an exception, preferably a custom one, is a way for the function to signal that something went wrong. The calling function (or app, or anything higher) can then catch this exception and react appropriately in the context where it knows what to do (exit, return an error code, log and ignore, etc).

Reply:
That makes sense. Working it out.